iOS build broken

This sort of thing will be caught more easily once I hit https://github.com/mapbox/mapbox-gl-cocoa/issues/28 so that iOS build bugs are caught upon commit. 

```
In file included from /tmp/gl/src/mbgl/map/tile_parser.cpp:15:
In file included from src/mbgl/text/collision.hpp:16:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:43:
In file included from /usr/local/include/boost/geometry/strategies/strategies.hpp:74:
In file included from /usr/local/include/boost/geometry/strategies/agnostic/relate.hpp:14:
In file included from /usr/local/include/boost/geometry/algorithms/detail/relate/relate.hpp:53:
/usr/local/include/boost/geometry/algorithms/detail/relate/linear_areal.hpp:300:81: error: implicit conversion loses integer precision: 'long' to 'const int' [-Werror,-Wshorten-64-to-32]
                        int const next_ring_index = prev_seg_id_ptr->ring_index + 1;
                                  ~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/local/include/boost/geometry/algorithms/detail/relate/linear_areal.hpp:371:73: error: implicit conversion loses integer precision: 'long' to 'const int' [-Werror,-Wshorten-64-to-32]
                int const next_ring_index = prev_seg_id_ptr->ring_index + 1;
                          ~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
In file included from /tmp/gl/src/mbgl/map/tile_parser.cpp:15:
In file included from src/mbgl/text/collision.hpp:16:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:48:
In file included from /usr/local/include/boost/geometry/algorithms/buffer.hpp:32:
In file included from /usr/local/include/boost/geometry/algorithms/detail/buffer/buffer_inserter.hpp:27:
In file included from /usr/local/include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp:42:
In file included from /usr/local/include/boost/geometry/algorithms/detail/occupation_info.hpp:21:
/usr/local/include/boost/geometry/algorithms/detail/get_left_turns.hpp:249:43: error: implicit conversion loses integer precision: 'unsigned long' to 'int' [-Werror,-Wshorten-64-to-32]
            previous_index = cluster_size - 1;
                           ~ ~~~~~~~~~~~~~^~~
3 errors generated.
```

--- a/src/mbgl/text/collision.hpp
+++ b/src/mbgl/text/collision.hpp
@@ -7,9 +7,9 @@
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wunused-variable"
 #pragma GCC diagnostic ignored "-Wshadow"
-#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
 #ifdef __clang__
 #pragma GCC diagnostic ignored "-Wdeprecated-register"
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
 #else
 #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"

--- a/src/mbgl/text/collision.hpp
+++ b/src/mbgl/text/collision.hpp
@@ -7,6 +7,7 @@
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wunused-variable"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
 #ifdef __clang__
 #pragma GCC diagnostic ignored "-Wdeprecated-register"
 #else

